### PR TITLE
Add wandb logger

### PR DESCRIPTION
**Pull request by nightingal3**
We wanted to add a wandb logger to make it easier to share progress on runs. I noticed that [megatron-LLM already had one](https://github.com/epfLLM/Megatron-LLM/blob/1b06b129fa463b7bfce88ef49e2082f8df00c7fa/megatron/wandb_logger.py#L4) so I integrated it here (this is mostly just a port of that).

I haven't used tensorboard so I'm not sure if this interferes with logging there, let me know if it does.

**To test**: run the training script `pretrain_gpt.py` with any existing arguments. and add the arguments
```
    --wandb_logger \
    --wandb_entity <your wandb username> \
    --wandb_id <run_id> \
    --wandb_api_key <your wandb api key>
```
You should see a number of panels in wandb. I had to separate the time intervals (e.g. `tokens/` `steps/` etc) to get the logger to work so I'm not sure if this is the cleanest way to do so (since the person adding a new metric has to remember to do this). The wandb logger should ignore things that aren't prepended with the interval type.

### CODE REVIEW

**Review by CoderPat — APPROVED** [review]
LGTM
Thanks @nightingal3 ! This is really useful (we have just been using tensorboard)
